db/index.go

```diff
@@ -270,33 +278,46 @@ func (i *collectionSimpleIndex) deleteDocIndex(
 	return i.deleteIndexKey(ctx, txn, key)
 }
 
+// hasIndexKeyNilField returns true if the index key has a field with nil value
+func hasIndexKeyNilField(key *core.IndexDataStoreKey) bool {
+	const cborNil = 0xf6
+	for i := range key.FieldValues {
+		if len(key.FieldValues[i]) == 1 && key.FieldValues[i][0] == cborNil {
```

Comment:
suggestion: This line earlier in the review threw me quite a bit, and it is repeated here.  It might be worth adding an `IsNill` func to `core.IndexDataStoreKey` with a bit of documentation.  

suggestion: It is also not great that we are having to duplicate the `cborNil` const in a few places. Can I suggest moving it to a shared place and making it accessible, and renaming it to something that does not include cbor in the name, so if it being cbor related changes we don't have to update a whole bunch of references to it (it being cbor also seems irrelevant to the referencing location).

Reply:
I share your concerns, I had the same thought. 
But I'm currently working on encoding on another branch. All this will be gone anyway.
That's why I didn't bother too much with this part.
If you still think I should do this in this PR, I can do it.

Reply:
Ah okay nice!  If you are confident it will be removed soon I'm happy for it to remain as-is here :)